benchmark_data_loading/dataset_helpers.py

```diff
@@ -182,10 +185,22 @@ def make_ffcv_dataloader(*, root, transforms, encoded):
     )
 
 
+def _extract_wds_archive_size(web_dataset):
+    # This function is a crutch since we need to know the archive_size for intra-archive shuffling, but this is not
+    # available in with_DL.
+    archive_paths = web_dataset.pipeline[0].urls
+    archive_name = Path(archive_paths[0]).name
+    return int(archive_name.split("_")[1])
```

Comment:
I'm tempted to cheat and make `archive_size` part of the `args` global var.... thoughts?

Reply:
I'll go for it.